server/broadcast.go

```diff
@@ -412,14 +423,25 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string)
 				}()
 			}
 
+			// Store URLs for the verifier. Be aware that the segment is
+			// already within object storage  at this point, whether local or
+			// external. If a client were to ignore the playlist and
+			// preemptively fetch segments, they could be reading tampered
+			// data. Not an issue if the delivery protocol is being obeyed.
+			segHashLock.Lock()
+			segURLs[i] = url
+			segHashLock.Unlock()
+
 			if monitor.Enabled {
 				monitor.TranscodedSegmentAppeared(nonce, seg.SeqNo, sess.Profiles[i].Name)
 			}
+			/* Should we do this conditionally in the presence of the verifier? Otherwise playlist insertions are blocked until all segments are available, which seems worse - but is necessary for verification
```

Comment:
Thought about this and I'd really prefer not to have two code paths that do the same thing depending on which mode the node is running in. That can get very confusing and error-prone very quickly, and I would really like to avoid setting that precedent here.

While we do have a lot of different operating modes on the node, the composition has been rather clear-cut so far: a bit of additional logic might execute for on-chain mode, or an orchestrator may invoke a remote transcoder in standalone mode, but there aren't really cases AFAIK where we execute the same logic in different places, depending on the mode. So that's a slope I'd like to avoid going down as much as possible.

Removed this bit here: https://github.com/livepeer/go-livepeer/pull/1096/commits/55a8d8b683a0dde032081be9201ddae5113f6aa7